utils/cache/blob_memory_cache.go

```diff
@@ -134,3 +134,40 @@ func (c *BlobMemoryCache) TotalBytes() int64 {
 	defer c.mu.RUnlock()
 	return c.totalSize
 }
+
+// GetExpiredEntries returns names of entries older than TTL.
+// Uses RLock for minimal contention (allows concurrent reads).
+func (c *BlobMemoryCache) GetExpiredEntries(now time.Time, ttl time.Duration) []string {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	var expiredNames []string
+
+	for name, entry := range c.entries {
+		if now.Sub(entry.CreatedAt) > ttl {
+			expiredNames = append(expiredNames, name)
+		}
+	}
+
+	return expiredNames
+}
+
+// RemoveBatch removes multiple entries atomically.
+// Uses Lock for batch deletion (single lock acquisition for entire batch).
+func (c *BlobMemoryCache) RemoveBatch(names []string) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	reclaimedBytes := int64(0)
+
+	for _, name := range names {
+		entry, exists := c.entries[name]
+		if !exists {
+			continue
+		}
```

Comment:
```suggestion
		if !exists {
		    // should never happen, but check just in case
			continue
		}
```

With the RW lock, this should be impossible, no? Is it a "just-in-case" check? If yes, can we add that comment?

Reply:
Or is it possible if
1. the cleanup worker gets the list of expired files
2. the drain workers evict the file
3. RemoveBatch is called